src/middlewared/middlewared/utils/origin.py

```diff
@@ -148,6 +149,30 @@ def secure_transport(self) -> bool:
         # By default assume that transport is insecure
         return False
 
+    def ppids(self) -> set[int]:
+        if self.pid is None:
+            return set()
+
+        pid = self.pid
+        ppids = set()
+        while True:
+            try:
+                with open(f"/proc/{pid}/status") as f:
+                    status = f.read()
```

Comment:
Instead of reading the entire file into memory for each parent process, we can iterate over the file line by line and only grab the line we want. We also don't need a regex for this file.
```suggestion
                    for line in filter(lambda x: x.startswith('PPid:', f):
                        try:
                            ppid = int(line.split(':')[1].strip())
                            if ppid <= 1:
                                break
                            else:
                                ppids.add(ppid)
                        except ValueError:
                            continue (or break?)
```